sdk/simulator-service/src/implementation/simulator-engine/reducer/paybackWithdrawReducer.ts

```diff
@@ -7,12 +7,10 @@ export function paybackWithdrawReducer(
   step: steps.PaybackWithdrawStep,
   state: ISimulationState,
 ): ISimulationState {
+  const paybackAmount = getValueFromReference(step.inputs.paybackAmount)
   const withdrawAmount = getValueFromReference(step.inputs.withdrawAmount)
-  const afterPayback = addBalance(withdrawAmount, state.balances)
-  const afterWithdraw = subtractBalance(
-    getValueFromReference(step.inputs.withdrawAmount),
-    afterPayback,
-  )
+  const afterPayback = subtractBalance(paybackAmount, state.balances)
+  const afterWithdraw = addBalance(withdrawAmount, afterPayback)
```

Comment:
These two looks suspicious (`afterPayback` & `afterWithdraw`). Could you double check whether usage of `substractBalance` and `addBalance` are used as intended?

Reply:
Yeah, I think it was a bug in the previous iteration. I've compared with `depositBorrowReducer` and it seems that we use `subtractBalance` when moving tokens from DPM to the protocol, and `addBalance` for the opposite. So both payback and deposit should subtract (DPM -> Protocol) and withdraw and borrow should add (Protocol -> DPM). Does it make sense?